src/algokit/cli/compile.py

```diff
@@ -0,0 +1,93 @@
+import logging
+from collections.abc import MutableMapping, Sequence
+from typing import Any
+
+import click
+
+from algokit.core.compile.python import find_valid_puyapy_command
+from algokit.core.proc import run
+
+logger = logging.getLogger(__name__)
+
+
+class CompileGroup(click.Group):
```

Comment:
There is a lot here this isn't fully utilized, other samples of overwriting the group class usually inject dynamic commands or take care of aliasing. In this case if we want to have both py/python to be accessible there are a few simpler options:

1.  Using a simplified custom group class with only get_command overload
```
class AliasedGroup(click.Group):
    def get_command(self, ctx: click.Context, cmd_name: str) -> click.Command | None:
        rv = click.Group.get_command(self, ctx, cmd_name)
        if rv is not None:
            return rv

        if cmd_name == "py":
            return click.Group.get_command(self, ctx, "python")
          
         ... and so on for any ts -> typescript and other shorthands. 

        return None
```

In this case docstring in the python command need to include ref that its also callable from py:
```
Compile Python contract(s) to TEAL with PuyaPy. Additionally available via `algokit compile py` shorthand alias.
```
       
2. Adding duplicated python command to the compile group manually:
    
  ```
@click.command(
    "python",
    short_help="Compile Python to TEAL with PuyaPy",
    help="Compile Python to TEAL with PuyaPy, review https://github.com/algorandfoundation/puya for usage",
    context_settings={
        "ignore_unknown_options": True,
    },
    add_help_option=False,
)
@click.argument("puya_args", nargs=-1, type=click.UNPROCESSED)
@click.pass_context
def compile_py_command(context: click.Context, puya_args: list[str]) -> None:
    """
    Compile Python contract(s) to TEAL with PuyaPy
    """
    version = str(context.obj["version"]) if context.obj["version"] else None

    puya_command = find_valid_puyapy_command(version)

    run_result = run(
        [
            *puya_command,
            *puya_args,
        ],
    )
    click.echo(run_result.output)

    if run_result.exit_code != 0:
        click.secho(
            "An error occurred during compile. Ensure supplied files are valid PuyaPy code before retrying.",
            err=True,
            fg="red",
        )
        raise click.exceptions.Exit(run_result.exit_code)


compile_group.add_command(compile_py_command, "python")
compile_group.add_command(compile_py_command, "py")
  ```
  
  Last 2 lines achieve a simpler equivalent of what you currently have in CompileGroup class.